src/lib/utils/strong_type.h

```diff
@@ -47,9 +47,13 @@ class Strong_Base {
 
       constexpr explicit Strong_Base(T v) : m_value(std::move(v)) {}
 
-      T& get() { return m_value; }
+      constexpr T& get() & { return m_value; }
 
-      const T& get() const { return m_value; }
+      constexpr const T& get() const& { return m_value; }
+
+      constexpr T&& get() && { return std::move(m_value); }
+
+      constexpr const T&& get() const&& { return std::move(m_value); }
```

Comment:
This was inspired by https://github.com/randombit/botan/pull/4175#discussion_r1671759180 and it simplifies the code of `strong_type_unwrap()` significantly.